lib/sequent/core/persistors/replay_optimized_postgres_persistor.rb

```diff
@@ -346,7 +346,10 @@ def clear
         private
 
         def cast_value_to_column_type(clazz, column_name, record)
-          Sequent::ApplicationRecord.connection.type_cast(record[column_name.to_sym], @column_cache[clazz.name][column_name])
+          data = clazz.arel_table.type_cast_for_database(column_name, record[column_name.to_sym])
+          return data unless data.respond_to?(:encoder)
```

Comment:
Is there a case when this is not the case?

Reply:
For strings it returns the string right away instead of the encoder.